scrapy/commands/genspider.py

```diff
@@ -61,6 +62,13 @@ def run(self, args, opts):
             raise UsageError()
 
         name, domain = args[0:2]
+
+        parsed_url = urlparse(domain)
+
+        # Automatically it is prepended 'http' if the scheme is missing
+        if not parsed_url.scheme:
+            parsed_url = urlparse('http://' + domain)
```

Comment:
Since you already have a named tuple with the parsed URL, you can use the [`_replace` method](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace) to update the schema (see also https://docs.python.org/3/library/urllib.parse.html?highlight=_replace#urllib.parse.urlparse)

Reply:
For what I understand, I accept your changes but still, I need to add 2 more: to remove the unnecessary comment and to use the `_replace` method to update the schema. Sorry If I am being repetitive but it's a little bit confusing what do I have to do after your reviewing.